scripts/generate_tutorial_readmes.py

```diff
@@ -138,21 +144,23 @@ def main():
 
 def load_youtube_playlist_urls():
     """Create a mapping from day code to youtube link based on text file."""
-    with open("tutorials/youtube_playlists.txt") as f:
-        lines = filter(bool, f.read().split("\n"))
-    return dict(tuple(line.split()) for line in lines)
+    with open('tutorials/materials.yml') as fh:
+        materials = yaml.load(fh, Loader=yaml.FullLoader)
```

Comment:
I don't think you imported yaml anywhere? Please add it to the requirements file as well. (Then you won't need to pip install it in the github workflows).